### PR TITLE
Simplify prefix

### PR DESCRIPTION
**Pull request by prashanthpai**

### CODE REVIEW

File: store/store.go [hunk 1]
```diff
@@ -77,27 +53,17 @@ func (s *GDStore) InitPrefix(p string) error {
...
-
 			if e := s.Put(p, nil, &store.WriteOptions{IsDir: true}); e != nil {
-				log.WithFields(log.Fields{
-					"preifx": p,
-					"error":  e,
-				}).Error("error initing prefix")
-
 				return e
+			} else {
+				log.WithField("prefix", p).Debug("Prefix not found. Created prefix.")
```
Comment (kshlm) [inline store/store.go]: Split this log. Log "Prefix not found" when you enter the case, and "Created prefix" only on success.

File: gdctx/global.go [hunk 2]
```diff
@@ -71,7 +71,12 @@ func doInit() {
 
 	Rest = rest.New()
 
-	initStore()
+	if Restart == true {
```
Comment (kshlm) [inline gdctx/global.go]: Couldn't this just be `InitStore(!Restart)`. Add a comment clarifying what's being done above this, and everyones happy.

File: gdctx/store.go [hunk 3]
```diff
@@ -15,14 +19,25 @@ func RegisterStorePrefix(prefix string) {
 }
 
 // InitStore is to initialize the store
-func initStore() {
-	Store = store.New(Restart)
+func InitStore(initPrefix bool) {
+	Store = store.New()
+
+	if initPrefix == true {
```
Comment (kshlm) [inline gdctx/store.go]: The linter should have complained about this. You don't need to check for `boolval == true` for an `if` statement. Keep it `if initPrefix {`.

Comment (kshlm) [inline gdctx/store.go]: On why this wasn't caught, `make verify` uses gometalinter, which I've set to ignore warnings for everything including golint. I'll probably change it some time later.

File: gdctx/store.go [hunk 4]
```diff
@@ -15,14 +19,25 @@ func RegisterStorePrefix(prefix string) {
 }
 
 // InitStore is to initialize the store
-func initStore() {
-	Store = store.New(Restart)
+func InitStore(initPrefix bool) {
+	Store = store.New()
```
Comment (kshlm) [inline gdctx/store.go]: Begin variable names with lowercase, unless you are exporting the variable. Infact, capitalize names of variables, types, functions etc. only if you want to export them. If you are only using these within this package, keep them all in lowercase and unexported.

Comment (prashanthpai) [inline gdctx/store.go]: In this particular case, `Store` variable is exported and consumed by other packages.

Comment (kshlm) [inline gdctx/store.go]: Cool! It's fine then.

**Pull request merged**